src/main/scala/uk/gov/hmrc/bobby/domain/bobby.scala

```diff
@@ -18,28 +18,57 @@ package uk.gov.hmrc.bobby.domain
 
 import sbt.librarymanagement.ModuleID
 
-sealed trait BobbyResult {
-  def failed: Boolean
+sealed trait BobbyResult extends Product with Serializable {
   def rule: Option[BobbyRule]
-  def name: String
+
+  final def name: String =
+    this match {
+      case BobbyViolation(_) => BobbyViolation.tag
+      case BobbyWarning(_)   => BobbyWarning.tag
+      case BobbyExemption(_) => BobbyExemption.tag
+      case BobbyOk           => BobbyOk.tag
+    }
+
+  protected def ordering: Int
```

Comment:
I think this would be easier to scan if defined on the trait as an exhaustive match as with `name` above. Thoughts?